### PR TITLE
Move validation commands into group

### PR DESCRIPTION
**Pull request by samuelallan72**
[BSENG-2024](https://warthogs.atlassian.net/browse/BSENG-2024)

<details>
<summary>Examples for reference:</summary>

```
ubuntu@serverstack-sunbeam-0:~$ openstack.sunbeam validation
Usage: sunbeam validation [OPTIONS] COMMAND [ARGS]...

  Manage cloud validation functionality.

Options:
  -h, --help  Show this message and exit.

Commands:
  run         Run a set of tests on the sunbeam installation.
  test-lists  Get supported test lists for validation.


ubuntu@serverstack-sunbeam-0:~$ openstack.sunbeam validation run --help
Usage: sunbeam validation run [OPTIONS]

  Run a set of tests on the sunbeam installation.

Options:
  -s, --smoke               Run the smoke tests only. Equivalent to
                            --regex=smoke.  If --regex and --smoke are
                            provided, --regex is ignored.
  -r, --regex TEXT          A list of regexes, whitespace separated, used to
                            select tests from the list.
  -e, --exclude-regex TEXT  A single regex to exclude tests.
  -t, --serial              Run tests serially. By default, tests run in
                            parallel.
  -l, --test-list TEXT      Use a predefined test list. See `sunbeam
                            validation-lists` for available test lists.
  -h, --help                Show this message and exit.


ubuntu@serverstack-sunbeam-0:~$ openstack.sunbeam validation test-lists --help
Usage: sunbeam validation test-lists [OPTIONS]

  Get supported test lists for validation.

Options:
  -h, --help  Show this message and exit.



ubuntu@serverstack-sunbeam-0:~$ openstack.sunbeam configure -h
Usage: sunbeam configure [OPTIONS] COMMAND [ARGS]...

  Configure cloud with some sensible defaults.

Options:
  -a, --accept-defaults  Accept all defaults.
  -p, --preseed FILE     Preseed file.
  -o, --openrc FILE      Output file for cloud access details.
  -h, --help             Show this message and exit.

Commands:
  caas        Configure Cloud for Container as a Service use.
  validation  Configure validation plugin.


ubuntu@serverstack-sunbeam-0:~$ openstack.sunbeam configure validation
Config options available:

schedule: set a cron schedule for running periodic tests.  Empty disables.

Run with key=value args to set configuration values.
For example: sunbeam configure validation schedule="*/30 * * * *"



ubuntu@serverstack-sunbeam-0:~$ openstack.sunbeam configure validation schedule="foo"
Error: Exactly 5 columns must be specified for iterator expression.
ubuntu@serverstack-sunbeam-0:~$ openstack.sunbeam configure validation schedule="* * * * *"
Error: Cannot schedule periodic check to run faster than every 15 minutes.
ubuntu@serverstack-sunbeam-0:~$ openstack.sunbeam configure validation schedule
Error: syntax: key=value
ubuntu@serverstack-sunbeam-0:~$ openstack.sunbeam configure validation test=something
Error: test is not a supported config option

```

</details>

### CODE REVIEW

File: sunbeam-python/sunbeam/plugins/validation/plugin.py
Comment (samuelallan72) [inline sunbeam-python/sunbeam/plugins/validation/plugin.py:97]: This dataclass structure is set up to accomodate more config options in the future.  Unsure how premature this is.  Either way, it helps with a structure for validation and typing.

File: sunbeam-python/sunbeam/plugins/validation/plugin.py
Comment (samuelallan72) [inline sunbeam-python/sunbeam/plugins/validation/plugin.py:242]: There are a lot of user-facing strings here - please give suggestions for improvement so the help text provides a good UX. :)

File: sunbeam-python/sunbeam/plugins/validation/plugin.py
Comment (agileshaw) [inline sunbeam-python/sunbeam/plugins/validation/plugin.py:115]: I'm surprised to see syntax validation being handled by our implementation. Since click is built on top of argparse, and argparse has different action for arguments to know whether an argument allows multiple declaration, I would expect click to have this figured out on its side.

File: sunbeam-python/sunbeam/plugins/validation/plugin.py
Comment (agileshaw) [inline sunbeam-python/sunbeam/plugins/validation/plugin.py:131]: This also. It would make much more sense if this is handled by click or argparse internally.

File: sunbeam-python/sunbeam/plugins/validation/plugin.py
Comment (agileshaw) [inline sunbeam-python/sunbeam/plugins/validation/plugin.py]: To be clearer:
```suggestion
        help="Run the smoke tests only. Equivalent to --regex=smoke.  If both --regex and --smoke are provided, --regex is ignored.",
```

File: sunbeam-python/sunbeam/plugins/validation/plugin.py
Comment (agileshaw) [reply sunbeam-python/sunbeam/plugins/validation/plugin.py:242]: One idea would be to make `sunbeam configure validation` a subcommand and change `schedule` to argument/option. The message would not be too length if they are in the help message. It would also resolve my two other comments.

File: sunbeam-python/sunbeam/plugins/validation/plugin.py
Comment (agileshaw) [reply sunbeam-python/sunbeam/plugins/validation/plugin.py:115]: Now I see we are using key-value pairs as inputs instead of arguments. My suggestion below would save us from implementing this logic and delegate the responsibility to click and argparse.

**Review by agileshaw — CHANGES_REQUESTED** [review]
Thanks for the patch. As suggested inline, I think it would be easier and cleaner if we treat key-value pairs as argument instead of nargs.

File: sunbeam-python/sunbeam/plugins/validation/plugin.py
Comment (samuelallan72) [reply sunbeam-python/sunbeam/plugins/validation/plugin.py:115]: Unfortunately it doesn't appear that click or argparse supports `key=value` options.  They support `--key=value`, but we wanted a cleaner interface here, that was consistent with other tools like `juju config`.

**Review by agileshaw — APPROVED** [review]
After investigation, there's no native support in Python that supports `key=value` style arguments. So we can keep it the way it is.

File: sunbeam-python/sunbeam/plugins/validation/plugin.py
Comment (agileshaw) [reply sunbeam-python/sunbeam/plugins/validation/plugin.py:242]: We can keep it as is and wait for sunbeam's feedback when submitting to upstream

**Comment by agileshaw** [comment]
@rgildein Feel free to take a look at this patch if you'd like. But for the moment, since this is just to merge to my fork and Sunbeam team will be the one having final say on the CLI structure, I'm merging the PR

**Pull request merged**
